server/plugin.go

```diff
@@ -152,6 +168,21 @@ func (p *Plugin) GetClientForTeamsUser(teamsUserID string) (msteams.Client, erro
 		return nil, errors.New("not connected user")
 	}
 
+	if token.Expiry.Before(time.Now()) {
+		var err error
+		token, err = msteams.RefreshToken(p.GetURL()+"/oauth-redirect", p.getConfiguration().TenantID, p.getConfiguration().ClientID, p.getConfiguration().ClientSecret, token)
+		if err != nil {
+			return nil, err
+		}
+		userID, err := p.store.TeamsToMattermostUserID(teamsUserID)
```

Comment:
I understand that this is the "inverse" of what we do in `GetClientForUser`, but I wonder if we can simplify and reduce code duplication here if we just reimplement this method as something like the following?
```go
func (p *Plugin) GetClientForTeamsUser(teamsUserID string) (msteams.Client, error) {
        userID, err := p.store.TeamsToMattermostUserID(teamsUserID)
        if err != nil {
                return nil, err
        }
        
        return p.GetClientForUser(userID)
}
```

This does add an extra database lookup for this code path (unless we also happen to refresh), but feels more resilient. And maybe we can optimize that later by just returning more from the store in the first place allowing us to do the conversion instantly?

Reply:
Yes, that makes sense, let me change that.

Reply:
Done